packages/server/lib/browsers/firefox.ts

```diff
@@ -529,7 +536,7 @@ export async function open (browser: Browser, url: string, options: BrowserLaunc
     // user can overwrite this default with these env vars or --height, --width arguments
     MOZ_HEADLESS_WIDTH: '1280',
     MOZ_HEADLESS_HEIGHT: '721',
-  }) as LaunchedBrowser & { browserCriClient: BrowserCriClient}
+  }) as unknown as BrowserInstance
```

Comment:
I haven't seen this before, tricky.

Reply:
Yeah... I don't know the "right" way to change the type of an object, type-safely, without using `class ... extends ...`. If there even is a way without `as unknown`. Would be nice to eventually remove the need for this.

Reply:
This one seems a quick fix https://github.com/cypress-io/cypress/pull/23693

I think what you are referring to is something like

```ts
function foo<T extends SomeBaseClass>(arg: T): T {
  // 
}
```

Now as long as `arg` extends `SomeBaseClass` you should get type safety (error if you don't extend the base class).

Reply:
Maybe like this? https://www.typescriptlang.org/play?#code/IYIwzgLgTsDGEAJYBthjAgagUwBYEsVsEBvAKAUoVEhngQHNtF8ATBACgEoEAuBWvgB2DMmKoIA7rmzZkYfkICuAWxDYoFKrAD2Q2kvg6oHabPmLV6qD3ISJEAmAB0ZuRgC8Ume62UAvmSBZChoGADCwFAI2AAeENhCrBg4BESkfghsCF4A5LBRuZmJDMLY-Lly2PBQhLkIAD4IuQAOzFA6yEWZuvrQhhDGHCVlFVU1dY3NbdCdubaZEmBKbSYALFyLVI74LiNCxF772Jn+CEFioegIAEL4ANbEcQlJKXiEyMR2VNl5IA-YIoSFQ6VhyCogFSxepNXIgpRCCDAYQw5odYCsboSXoGIwmEFg5AQqGouE6BFIlFTXLozELexUZarDgAJk2DMoOxcBLkOQQPOQp3OwQAZgj4Pg9IwoLIIAAeAAqMXiiWSWHeRAAfBwWsAAJ7IHQY-gK+lIPRgTrYZyGhgcAAGCr1bSyGAAJCRdQajaxnGwzpJ8I4EB6vYaMa4fPIzso1BoEDoRd5zC57ZtggwZcwOAdJAhIiZKp8JrB5ptM7Kc9g83dHhwyRTkUIy2QgA

Also works with `interface` and plain objects ... not just limited to `class`.